zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -366,6 +367,10 @@ private void batchProcessing(final TypedRecord<?> initialCommand) {
 
       final var command = pendingCommands.removeFirst();
 
+      if (!command.equals(initialCommand)) {
+        currentSourceIndex = command.getKey();
```

Comment:
:x: This is not correct because it mixes two different things: we are interested in the _index_ of this command within the batch. This is different to the record _key_ which, for these purposes, is an arbitrary number unrelated to the index.

Reply:
Reverted.